server/publications/collections/products.js

```diff
@@ -494,11 +502,21 @@ Meteor.publish("Products", function (productScrollLimit = 24, productFilters, so
     });
   }
 
+  // if shops were requested in the product filter, limit the set to those
+  // which are active
```

Comment:
product filters allow for a set of shops. While some work had been done to validate the requested Shops, the filter is ultimately discarded by the `{ $in: activeShopIds }` selector.
This takes the intersection of requested shops and the active shops and returns those products.
the related test was yielding a false positive, which became apparent when I added more shops and products to the tests.